### PR TITLE
Bulk operations

### PR DESCRIPTION
**Pull request by Pidid**
@williamkapke 

I have added basic/partial support for bulk operations (`initializeOrderedBulkOp` and `initializeUnorderedBulkOp`).

Something to note is the difference between the **OrderedBulk** and **UnorderedBulk**. In my code, they are actually very similar. The reason for this is due to mongodb being vague on their implementation details in their docs. The most substantial difference I found was 

> With an unordered operations list, MongoDB can execute in parallel the write operations in the list and in any order.

So in the commited code, the **OrderedBulk** executes an operation one after another and in the order that it was given. The **UnorderedBulk** executes its operations in a more "parallel" manner (it really just calls each operation as fast as it can instead of waiting for the last one to finish).

Another difference mentioned in mongodb's documentation to consider for future changes to the bulk operation is

> When executing an ordered list of operations, MongoDB groups the operations by the operation type and contiguity; i.e. contiguous operations of the same type are grouped together. 

and 

> The sizes and grouping mechanics are internal performance details and are subject to change in future versions.

So to truly emulate how mongodb does **OrderedBulk** operations, we will need to refer to the actual source code (which I plan to visit and implement in the near future).

Functions that **have been implemented**:
`collection.initializeOrderedBulkOp()`
`collection.initializeUnorderedBulkOp()`
`Bulk.find()`
`Bulk.insert()`
`Bulk.execute()`
`Bulk.find.remove()`
`Bulk.find.removeOne()`
`Bulk.find.update()`
`Bulk.find.updateOne()`

Functions that **haven't been implemented**:
`Bulk.getOperations()`
`Bulk.find.collation()`
`Bulk.find.arrayFilters()`
`Bulk.find.replaceOne()`

References:
[initializeOrderedBulkOp](https://docs.mongodb.com/manual/reference/method/db.collection.initializeOrderedBulkOp/#db.collection.initializeOrderedBulkOp)

[initializeUnorderedBulkOp](https://docs.mongodb.com/manual/reference/method/db.collection.initializeUnorderedBulkOp)

### CODE REVIEW

**Comment by AngefloMusic** [comment]
Awesome I was actually going to implement this over the weekend, glad you already did!

**Comment by Pidid** [comment]
@AngefloMusic Thanks! There are still those 4 functions that haven't been implemented, so feel free to implement those if you'd like!

**Comment by AngefloMusic** [comment]
I'm going to take a look at this implementation today.

File: lib/bulk.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            }
+            return collection.deleteOne({_id: doc._id});
+          });
+        },
+      });
+
+      return this;
+    },
+    replaceOne: NotImplemented,
+    // replaceOne: function(replacement) {
```
Comment (AngefloMusic) [inline lib/bulk.js]: Can we remove this before merge, I don't want to put unfinished code into the project, maybe keep this on a separate branch for later.

File: lib/bulk.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
+module.exports = function Bulk(collection, ordered) {
+  var ops = [];
+  var executedOps = [];
+  var interface = {
```
Comment (AngefloMusic) [inline lib/bulk.js]: can we move away from using `interface` as a variable name to stay compatible with people who run in strict mode?

Comment (AngefloMusic) [reply lib/bulk.js]: #59 lists a solution of a readable abbreviation.

**Pull request merged**